Ex5/Ex5.java

```diff
@@ -0,0 +1,105 @@
+     *
+     * @param dictionary - a list of words
+     * @return - an ordered list of characters
+     */
+    public static List<Character> findAlphabetOrder(List<String> dictionary) {
+        Graph alphabetGraph = new Graph();
+        int index = 0;
+        List<String> updateDict = updateDictionary(dictionary, index);
+        List<List<String>> newDictionary = findSubDictionary(updateDict, index);
+
+        while (newDictionary != null) {
+            for (List<String> subDict : newDictionary) {
+                List<Character> subAlphabet = findAlphabetList(subDict, index);
+                alphabetGraph.updateGraph(subAlphabet);
+            }
+            index++;
+            updateDict = updateDictionary(updateDict, index);
+            newDictionary = findSubDictionary(updateDict, index);
+        }
+        return alphabetGraph.topologicalSort();
+    }
+
+    /**
+     * The function receives a dictionary (a list of words in lexicographic order) and an index, and return
+     * a list of dictionaries according to their prefixes
+     *
+     * @param dictionary - a list of words in lexicographic order
+     * @param index - the index of the char we check
+     * @return - a list of dictionaries
+     */
+    private static List<List<String>> findSubDictionary(List<String> dictionary, int index) {
+        List<List<String>> dictionaryList = new ArrayList<>();
+        List<String> subDictionary = new ArrayList<>();
+
+        for (int j = 0; j < dictionary.size(); j++) {
+            if (dictionary.get(j).length() <= index) {
+                continue;
+            }
+            if (j == 0) {
+                subDictionary.add(dictionary.get(j));
+                continue;
+            }
+            // check if the prefixes from zero to index are equals
+            if (dictionary.get(j).substring(0, index).equals(dictionary.get(j - 1).substring(0, index))) {
+                subDictionary.add(dictionary.get(j));
+            } else {
+                if (subDictionary.size() > 0) {
+                    dictionaryList.add(subDictionary);
+                }
+                subDictionary = new ArrayList<>();
+                subDictionary.add(dictionary.get(j));
+            }
+        }
+        if (dictionaryList.size() == 0 && subDictionary.size() == 0) {
+            return null;
+        }
+        if (subDictionary.size() > 0) {
+            dictionaryList.add(subDictionary);
+        }
+        return dictionaryList;
+    }
+
+    /**
+     * The function receives a dictionary and an index, and return an ordered list of the characters at the index
+     *
+     * * @param dictionary - a list of words in lexicographic order
+     * @param index - char index in the word
+     * @return - a list of ordered characters
+     */
+    private static List<Character> findAlphabetList(List<String> dictionary, int index) {
+        List<Character> alphabet = new LinkedList<>();

```

Comment:
In Java it is much better to use `ArrayList` everywhere, unless you have a very good reason to use `LinkedList`.
`LinkedList` has much bigger overhead and much worse performance (see comment above about using `get(index)`).